.github/workflows/update-versions.yml

```diff
@@ -73,56 +77,81 @@ jobs:
         cache-from: type=gha
         cache-to: type=gha,mode=max
 
-  update-versions:
+  create-version-update-pr:
+    if: github.event_name == 'push'
     runs-on: ubuntu-latest
-    needs: validate-tag
+    needs: [validate-tag, build-and-push-run-image]
     permissions:
       contents: write
+      pull-requests: write
 
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
       with:
         fetch-depth: 0
+        token: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: Switch to main branch
+    - name: Create update branch
       run: |
-        git checkout main
-        git pull origin main
+        git checkout -b update-versions-${{ needs.validate-tag.outputs.version }}
+        git push origin update-versions-${{ needs.validate-tag.outputs.version }}
 
     - name: Update buildpack.toml versions
       run: make update-buildpack-versions RELEASE_VERSION=${{ needs.validate-tag.outputs.version }}
 
     - name: Update builder.toml
       run: make update-builder-versions RELEASE_VERSION=${{ needs.validate-tag.outputs.version }}
 
-    - name: Commit version updates
+    - name: Commit and push version updates
       run: |
         git config user.name "github-actions[bot]"
         git config user.email "github-actions[bot]@users.noreply.github.com"
         git add .
         if ! git diff --staged --quiet; then
           git commit -m "chore(release): Update buildpack and run-image versions to ${{ needs.validate-tag.outputs.version }}"
-
-          # Move tag to new commit
-          git tag -f ${{ needs.validate-tag.outputs.version }}
-          git push origin main
-          git push origin ${{ needs.validate-tag.outputs.version }} --force
+          git push origin update-versions-${{ needs.validate-tag.outputs.version }}
         fi
 
+    - name: Create Pull Request
+      uses: actions/github-script@v7
```

Comment:
you could use https://github.com/marketplace/actions/create-pull-request - I believe that will also take care of branch creation and committing / pushing